src/main/java/com/commercetools/sync/products/utils/ProductUpdateActionUtils.java

```diff
@@ -566,6 +569,48 @@
             });
     }
 
+
+    /**
+     * Compares the {@link io.sphere.sdk.taxcategories.TaxCategory} references of an old {@link Product} and
+     * new {@link ProductDraft}. If they are different - return {@link SetTaxCategory} update action.
+     *
+     * <p>If the old value is set, but the new one is empty - the command will unset the tax category.
+     *
+     * @param oldProduct the product which should be updated.
+     * @param newProduct the product draft with new {@link io.sphere.sdk.taxcategories.TaxCategory} reference.
+     * @return An optional with {@link SetTaxCategory} update action.
+     */
+    @Nonnull
+    public static Optional<SetTaxCategory> buildSetTaxCategoryUpdateAction(
```

Comment:
It's kind of holly war, but i'd like to return as specific type as possible, so it is easier to test and verify the values if you need.

The convention i'm usually following:
_accept as wide as possible, return as narrow as possible_

Of course, narrow doesn't mean always too narrow. This example with list is more or less good example when wide return type is more conventional, and in case of Lists i'd always recommend to return lists (or even `Collection`), so you could change implementation always without interface change.

If some customers use returned `SetTaxCategory` in such a way so they rely on specific type - they should be notified ASAP if the new returned type is not `SetTaxCategory` any more in the new versions. This is better than force users to make explicit type casting if they need and access to returned update action properties, which will silently built and fail at runtime after the library update.